scripts/apply_clang_format.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python
+
+"""Invokes clang-format to the files that have been whitelisted for automatic formatting."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import hashlib
+import os
+import platform
+import stat
+import subprocess
+import sys
+
+import get_clang_format as gcf
+
+
+# This is the list of PyTorch directories that should be formatted.
+CLANG_FORMAT_DIRS = [
+    "torch/csrc/jit",
+]
+
+
+def apply_clang_format(pytorch_root, target_dirs):
+    """
+    Apply clang-format to all files in all target_dirs; target_dirs is traversed recursively.
+    """
+    # Construct the path to the clang-format binary.
+    clang_format_bin = os.path.join(pytorch_root, ".clang-format-bin", "clang-format")
+
+    # Check to make sure the clang-format binary exists.
+    if not os.path.exists(clang_format_bin):
+        print("clang-format binary not found")
+        return False
+
+    # This stack contains all directories that have not been scanned yet.
+    dir_stack = [os.path.join(pytorch_root, target_dir) for target_dir in target_dirs]
+
+    while dir_stack:
```

Comment:
check out `pathlib`, you can do things like `path.glob(**/*.cpp)` to get all cpp files recursively under `path`. And in general it provides a nice object-oriented API for filesystem access (`foo / bar` for path joining, `foo.exists()` for existence checking, etc.) Probably most of the filesystem manipulation in this script would be cleaner using it.

Reply:
I stole what the old script was doing using `os.walk`, let me know if that's not good enough.

Reply:
Using `pathlib` would be nice (we previously couldn't because it isn't available in Py2), but it's not that important.